lib/galaxy/managers/histories.py

```diff
@@ -207,6 +207,17 @@ def p_tag_filter(term_text: str, quoted: bool):
                 self.model_class.deleted == (true() if show_deleted else false())
             )
 
+        # Handle archived histories
+        if show_published:
+            # Someone published a history and then archived it, it should still be returned
+            stmt = stmt.where(or_(self.model_class.archived == false(), self.model_class.published == true()))
+        elif show_shared:
+            # Someone shared a history with the current user and then archived it, it should still be returned
+            stmt = stmt.where(or_(self.model_class.archived == false(), self.user_share_model.user == user))
+        else:
+            # By default, only return non-archived histories
+            stmt = stmt.where(self.model_class.archived == false())
+
```

Comment:
Can you add a `show_archived` parameter on the history index endpoint ? That would simplify the query here since you can just follow that parameter, and it's also less surprising ?

Reply:
Hmmm, I tried that, but it would be confusing if you just use `show_published`. You will get only a subset of public histories unless you combine them with `show_archived`. Is that desirable?

Reply:
I've increased the test coverage after including `show_archived` and this is the "confusing" case that I was mentioning:

https://github.com/galaxyproject/galaxy/pull/17856/files#diff-0e4282e792feb48bd4102022b4520bc3213f1afd5df0b78bbd440c20b01f617fR334-R337

It might be ok, the important part is that we should be able to see other users' (and our own) public histories regardless of whether they are archived or not as long as they are not deleted or purged, and that case is working with `show_own=false`.